apps/console/src/extensions/components/sms-providers/pages/sms-providers.tsx

```diff
@@ -0,0 +1,677 @@
+            if (!values?.key) {
+                error.key = t(
+                    "extensions:develop.smsProviders.form..twilio.validations.required"
+                );
+            }
+            if (!values?.secret) {
+                error.secret = t(
+                    "extensions:develop.smsProviders.form.twilio.validations.required"
+                );
+            }
+            if (!values?.sender) {
+                error.sender = t(
+                    "extensions:develop.smsProviders.form.twilio.validations.required"
+                );
+            }
+        } else if (state.selectedProvider === "VonageSMSProvider") {
+            if (!values?.key) {
+                error.key = t(
+                    "extensions:develop.smsProviders.form.vonage.validations.required"
+                );
+            }
+            if (!values?.secret) {
+                error.secret = t(
+                    "extensions:develop.smsProviders.form.vonage.validations.required"
+                );
+            }
+            if (!values?.sender) {
+                error.sender = t(
+                    "extensions:develop.smsProviders.form.vonage.validations.required"
+                );
+            }
+        } else {
+            if (!values?.key) {
+                error.key = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.secret) {
+                error.secret = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.sender) {
+                error.sender = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.providerURL) {
+                error.providerURL = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.contentType) {
+                error.contentType = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.payload) {
+                error.payload = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+            if (!values?.provider) {
+                error.key = t(
+                    "extensions:develop.smsProviders.form.custom.validations.required"
+                );
+            }
+        }
+        
+        return error;
+    };
+
+    const handleConfigurationDelete = async (deleteBeforeUpdate?: boolean): Promise<boolean> => {
+        !deleteBeforeUpdate && setIsDeleting(true);
+
+        return deleteSMSProviders()
+            .then(() => {
+                !deleteBeforeUpdate && handleDeleteSuccess();
```

Comment:
Can you please check the logic here with regard to the purpose of the function added?

Reply:
The logic looks okay. The function take the parameter `deleteBeforeUpdate`. If we need to delete the configurations before updating the new configurations, the parameter will be true. Since it is true, we will not show any notifications for delete actions to the user.

When we need to call this function only to delete the configurations, then we can pass `deleteBeforeUpdate` as false or call the function without passing any value. In that case, the notifications will be displayed to the user